apps/ui/src/components/ProposalTimeline.vue

```diff
@@ -0,0 +1,130 @@
+<script setup lang="ts">
+import { _t } from '@/helpers/utils';
+import { Proposal, Space } from '@/types';
+
+type ProposalTimelineValues = {
+  created?: number;
+  start: number;
+  min_end: number;
+  max_end: number;
+};
+
+type State = {
+  id: string;
```

Comment:
We could further enhance this by using

```
id: keyof typeof labels;
```

after moving the `const labels = {` to the top. Maybe also use `LABELS` to highlight that it's a constant